### PR TITLE
CB-20692 ed25519 does not seem to be supported for SSH access

### PR DESCRIPTION
**Pull request by lajosrodek**
* Add support for ed25519 public keys.
  * Updated validations in both `environment` and `core`.
  * Applicable only if the public key material is passed in directly. No change to the scenario when `publicKeyId` is being used (AWS only).
* Ed25519 is apparently not FIPS-compliant, so environment / cluster launch using such a public key will instantly fail on AWS GovCloud (the only platform where FIPS is enabled at the moment).
* Rename `RsaPublicKeyValidatorTest` to `OpenSshPublicKeyValidatorTest` to reflect the name of `OpenSshPublicKeyValidator`.
* `PublicKeyReaderUtil`:
  * Remove unused methods `loadOpenSsh()` and `decodeDSAPublicKey()`.
  * `readEdEcPoint()` is based on https://stackoverflow.com/questions/65780235/ed25519-in-jdk-15-parse-public-key-from-byte-array-and-verify.
* Some code cleanup.
* Testing:
  * Manual test with DL in two AWS regions (`eu-central-1` and `us-gov-west-1`).
  * Added new UT & adapted existing ones.

### CODE REVIEW

File: core/src/test/java/com/sequenceiq/cloudbreak/service/stack/StackServiceTest.java [hunk 1]
```diff
@@ -1,13 +1,15 @@
 package com.sequenceiq.cloudbreak.service.stack;
```
[the remark this answers is not in the document]
Comment (lajosrodek) [reply core/src/test/java/com/sequenceiq/cloudbreak/service/stack/StackServiceTest.java:1]: Good point. Fixed for both this one and `OpenSshPublicKeyValidatorTest`.

File: common/src/main/java/com/sequenceiq/cloudbreak/util/PublicKeyReaderUtil.java [hunk 2]
```diff
@@ -20,10 +23,14 @@ public final class PublicKeyReaderUtil {
...
+    private static final String SSH2_ED25519_KEY = "ssh-ed25519";
+
+    private static final int NUM_BYTES_ED25519 = 32;
+
     private PublicKeyReaderUtil() {
     }
 
-    public static PublicKey load(String key) throws PublicKeyParseException {
+    public static PublicKey load(String key, boolean fipsEnabled) throws PublicKeyParseException {
         int c = key.charAt(0);
```
Comment (bergerdenes) [inline common/src/main/java/com/sequenceiq/cloudbreak/util/PublicKeyReaderUtil.java:34]: dead store?

File: common/src/test/java/com/sequenceiq/cloudbreak/util/PublicKeyReaderUtilTest.java [hunk 3]
```diff
@@ -1,21 +1,208 @@
...
-        Assertions.assertTrue(rsaKey.getAlgorithm().equals("RSA"));
+                "sc= test@test.local", false);
+        assertEquals("RSA", rsaKey.getAlgorithm());
+        if (rsaKey instanceof RSAPublicKey rsaPublicKey) {
+            assertThat(rsaPublicKey.getPublicExponent().toString()).isEqualTo("65537");
+            assertThat(rsaPublicKey.getModulus().toString()).isEqualTo(
+                    "15716563600099515211328423611954225824916533615230691376921478986333060601" +
+                    "57552226919808527872386330392237985485200227383711345332616167018657122816541835915" +
+                    "10152175666741773856946951958086291801538991290666845716385778192871160342792145141" +
+                    "097809456369701048006656617692262816450908563750970503449532801209031"
```
Comment (bergerdenes) [inline common/src/test/java/com/sequenceiq/cloudbreak/util/PublicKeyReaderUtilTest.java:35]: Hint: Might use """ textblock quote

File: core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    @Test
+    public void validRsaPublicKeyWillNotFailWhenMissingUser() {
+        underTest.validate(
+                        "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCasJyap4swb4Hk4xOlnF3OmKVwzmv2e053yrtvcUPaxCeboSltOBReuT"
+                                + "QxX+kYCgKCdtEwpIvEDXk16T6nCI4tSptAalFgpUWn+JOysCuLuWnwrk6mSKOzEiPYCrB54444mDY6rbBDSRuE/V"
+                                + "UYQ/yi0imocARlOiFdPRlZGTN0XGE1V8LSo+m0oIzTwBKn58I4v5iB4ZUL/6adGXo7dgdBh/Fmm4uYbgrCZnL1EaK"
+                                + "pMxSG76XWhuzFpHjLkRndz88ha0rB6davag6nZGdno5IepLAWg9oB4jTApHwhN2j1rWLN2y1c+pTxsF6LxBiN5rsY"
+                                + "KR495VFmuOepLYz5I8Dn", false
```
Comment (bergerdenes) [inline core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java:40]: Hint: Might use """ textblock quote

File: core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    @Test
+    public void inValidPublicKeyWillFailWhenMissingSshRsa() {
+        BadRequestException badRequestException = assertThrows(BadRequestException.class, () -> underTest.validate(
+                "AAAAB3NzaC1yc2EAAAADAQABAAABAQCasJyap4swb4Hk4xOlnF3OmKVwzmv2e053yrtvcUPaxCeboSltOBReuT"
+                + "QxX+kYCgKCdtEwpIvEDXk16T6nCI4tSptAalFgpUWn+JOysCuLuWnwrk6mSKOzEiPYCrB54444mDY6rbBDSRuE/V"
+                + "UYQ/yi0imocARlOiFdPRlZGTN0XGE1V8LSo+m0oIzTwBKn58I4v5iB4ZUL/6adGXo7dgdBh/Fmm4uYbgrCZnL1EaK"
+                + "pMxSG76XWhuzFpHjLkRndz88ha0rB6davag6nZGdno5IepLAWg9oB4jTApHwhN2j1rWLN2y1c+pTxsF6LxBiN5rsY"
+                + "KR495VFmuOepLYz5I8Dn sequence-eu", false
```
Comment (bergerdenes) [inline core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java:62]: Hint: Might use """ textblock quote

**Review by bergerdenes — APPROVED** [review]

File: common/src/test/java/com/sequenceiq/cloudbreak/util/PublicKeyReaderUtilTest.java (hunk 3 above)
Comment (lajosrodek) [reply common/src/test/java/com/sequenceiq/cloudbreak/util/PublicKeyReaderUtilTest.java:35]: It is a single line without linebreaks. :smile:

File: core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    @Test
+    public void validRsaPublicKeyWillNotFail() {
+        underTest.validate(
+                        "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCasJyap4swb4Hk4xOlnF3OmKVwzmv2e053yrtvcUPaxCeboSltOBReuT"
+                                + "QxX+kYCgKCdtEwpIvEDXk16T6nCI4tSptAalFgpUWn+JOysCuLuWnwrk6mSKOzEiPYCrB54444mDY6rbBDSRuE/V"
+                                + "UYQ/yi0imocARlOiFdPRlZGTN0XGE1V8LSo+m0oIzTwBKn58I4v5iB4ZUL/6adGXo7dgdBh/Fmm4uYbgrCZnL1EaK"
+                                + "pMxSG76XWhuzFpHjLkRndz88ha0rB6davag6nZGdno5IepLAWg9oB4jTApHwhN2j1rWLN2y1c+pTxsF6LxBiN5rsY"
+                                + "KR495VFmuOepLYz5I8Dn sequence-eu", false
```
[the remark this answers is not in the document]
Comment (lajosrodek) [reply core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java:29]: It is a single line without linebreaks. :smile:

File: core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java (hunk 4 above)
Comment (lajosrodek) [reply core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java:40]: It is a single line without linebreaks. :smile:

File: core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java (hunk 5 above)
Comment (lajosrodek) [reply core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java:62]: It is a single line without linebreaks. :smile:

**Pull request merged**